### ISSUE TITLE
Default value for optional readers

### ISSUE THREAD

**Issue by vkostyukov** [finagle/finch#186]
I was wondering if it might be useful. The idea is to instead of

```scala
val p: RequestReader[String] = OptionalParam("p") map { _.getOrElse("n/a") }
```

to allow users write something like this

```scala
val p: RequestReader[String] = OptionalParam("p") withDefault "n/a"
```

We can also use `orElse` instead of `withDefault`.

**Comment by vkostyukov** [comment]
I'm putting this into the next milestone.

**Comment by jenshalm** [comment]
I think that would be quite nice to have. If the value passed to the function is a simple value, `withDefault` works better as a name, as the expected signature for an `orElse` method would have an `Option[T]` param instead. For maximum flexibility you could have both:

```scala
def withDefault[B >: A](default: ⇒ B): RequestReader[Option[B]]
def orElse[B >: A](alternative: ⇒ Option[B]): RequestReader[Option[B]]
```

**Comment by vkostyukov** [comment]
Nice! Let's plan this for the 0.6.0 release.

**Comment by vkostyukov** [comment]
Fixed in #186.
